x/staking/keeper/delegation.go

```diff
@@ -643,9 +643,22 @@ func (k Keeper) Delegate(
 	// validator.Tokens
 	lastTotalPower := k.GetLastTotalPower(ctx)
 	maxVotingPowerEnforcementThreshold := k.MaxVotingPowerEnforcementThreshold(ctx)
-	if lastTotalPower.GTE(maxVotingPowerEnforcementThreshold) {
-		newVotingPowerRatio := sdk.NewDecFromInt(validator.Tokens.Add(bondAmt)).Quo(sdk.NewDecFromInt(lastTotalPower))
-		if newVotingPowerRatio.GT(k.MaxVotingPowerRatio(ctx)) {
+
+	// 1 power = Bond Amount / Power Reduction
+	validatorAddtionalPower := bondAmt.Quo(k.PowerReduction(ctx))
+	newTotalPower := lastTotalPower.Add(validatorAddtionalPower)
+
+	// If it's beyond genesis then enforce power ratio per validator if there's more than maxVotingPowerEnforcementThreshold
+	if newTotalPower.GTE(maxVotingPowerEnforcementThreshold) && ctx.BlockHeight() > 0 {
```

Comment:
Also check that the new power will be more than the enforcement or else the first delegation can be arbitrarily large

Reply:
Also don't check if this is genesis block as gentx will fail since delegate is called per validator sequentially , and the first couple will have more power ratio naturally